tests/integration_tests/functional/test_api.py

```diff
@@ -1184,10 +1185,26 @@ def test_api_version(test_microvm_with_api):
     assert 'firecracker_version' in postboot_response.json()
     # Validate VM version post-boot is the same as pre-boot.
     assert preboot_response.json() == postboot_response.json()
+
+    test_utils.configure_git_safe_directory()
     # Check that the version is the same as `git describe --dirty`.
-    out = subprocess.check_output(['git', 'describe', '--dirty']).decode()
-    # Skip the "v" at the start and the newline at the end.
-    assert out.strip()[1:] == preboot_response.json()['firecracker_version']
+    # Abbreviated to post-tag commit metadata
+    out = subprocess.check_output(['git', 'describe',
+                                   '--dirty', '--abbrev=0']).decode()
+
+    # Skip the "v" at the start
+    tag_version = out[1:]
+    # Strip the metadata appended to the tag
+    if out.find('-') > -1:
+        tag_version = tag_version[:tag_version.index('-')]
+    else:
+        # Just strip potential newlines
+        tag_version = tag_version.strip()
```

Comment:
```suggestion
    [_, *tag_version] = out
    if i := tag_version.find('-') != -1:
        tag_version = tag_version[:i]
    else:
        tag_version = tag_version.strip()
```
This presumes we can always simply skip the first character.
https://blog.teclado.com/destructuring-in-python/ and https://realpython.com/python-walrus-operator/ might be useful.

Reply:
I don't think the comparisons count here is critical for test code, and the readability is a bit easier laid out more in my opinion.

Reply:
I can understand that `i := tag_version.find('-') != -1:` is a little esoteric (perhaps we should record this decision to abstain from walrus operators in a style guide).

If you want to avoid this I would suggest instead:
```pyth
    [_v, *tag_version] = out
    i = tag_version.find('-')
    if i != -1:
        tag_version = tag_version[:i]
    else:
        tag_version = tag_version.strip()
```
Even though I have seen it before I personally find the `find` into `index` to be somewhat misleading from a readability perspective.

I also understand `[_v, *tag_version] = out` may raise some contention. Here I would strongly advocate for it as it helps clarifies we are dropping and not using the respective first character `_v` (I think de-structuring syntax can appear initially odd but helps readability (and safety) when you get used to it).